src/core.rs

```diff
@@ -47,11 +44,8 @@ impl Core {
             _ => flags.color.theme.clone(),
         };
 
-        let icon_theme = match (tty_available, flags.icons.when, flags.icons.theme) {
-            (_, IconOption::Never, _) | (false, IconOption::Auto, _) => icon::Theme::NoIcon,
-            (_, _, IconTheme::Fancy) => icon::Theme::Fancy,
-            (_, _, IconTheme::Unicode) => icon::Theme::Unicode,
-        };
+        let icon_when = flags.icons.when;
```

Comment:
Any reason to move this logic to inside `Icons::new`?

Reply:
the logic here is all about how to set icons, it should be handled inside the Icons package